### ISSUE TITLE
Feature Request: Justified (flush left and flush right) text alignment

### ISSUE THREAD

**Issue by rehanog** [koreader/koreader#59]
Hi,

Many thanks for the great work. Wonderful to be able to read epubs.

Feature request : could you provide a menu option that enables fully justified text alignment, with flush left and flush right text as in: http://en.wikipedia.org/wiki/Justification_(typesetting)

Many thanks,

Rehan

**Comment by retrue** [comment]
Hi.
This issue is from four years ago and closed but I would like to reopen it.
Nickel has an option to set up justified text. I would like to see that implemented too in KOReader. Imho the reason given by chrox is not good. For example, you cannot change the typefont in an epub when it has been designed in a way that changing it is only possible if you disable internal estyles, but most times it is not blocked andyou can. That happens too with text alignment. So, where text alignment is not blocked in the epub I would welcome an option to change it without disabling internal styles.
Thanks.

**Comment by felipevaz** [comment]
I second that. Left-justified alignment is a very basic function that *every* modern reader with much less capabilities than KOreader has, and there's a reason for that. Typographic designers know that left-aligned text is WAY more readable when either 1) there's no proper hyphenation; 2) if columns are somewhat narrow; 3) if words are large (think German), and each of these potentialize each other. In all of these very common situations, justified text creates large distracting gaps between words and it adds to eye strain. More info on these and other problems relating to justified text is available on the following links:
https://en.wikipedia.org/wiki/Typographic_alignment#Problems_with_justification
https://www.futurehosting.com/blog/why-does-justified-text-have-such-a-bad-reputation-on-the-web/
https://www.fonts.com/content/learning/fontology/level-2/making-type-choices/justified-vs-rag-right

In the publishing industry, justified text is *only* used when the designer is sure these problems won't arise, otherwise flush left is used so that the kerning of the fonts and appropriate space between words is guaranteed. For some reason, people culturally believe justified is "better", but most times it is actually bad design (see item 6 here: https://medium.com/@meetchopz/10-bad-typography-habits-that-scream-amateur-8bac07f9c041). In paper there WAS one more reason for justified text: the transparency of the paper caused the text from the other side of the paper to be seen, creating visual noise on the right edge of the text block, but there's no such issue on electronic devices.

It is clear to me that this is customizable by advanced users by tweaking CSS files, but from a designer's POV, there no justification (pun intended) for that. 

TLDR; would you developers be so kind as to reopen this issue and implement this in KOreader? Typographers and designers will be forever thankful!

**Comment by Frenzie** [comment]
I completely agree with you and I don't think justified makes for the best default, but you're replying to an issue from many years ago. In 2020 KOReader (or rather, 2018 KOReader and up), go to the document menu → style tweaks → text → text alignment. Hold to make an option the default.

<img src=https://user-images.githubusercontent.com/202757/72247945-4db29e80-35f6-11ea-9a16-438928c5ed37.png width=45%><img src=https://user-images.githubusercontent.com/202757/72247946-4e4b3500-35f6-11ea-8d73-7179142eeb84.png width=45%>
<img src=https://user-images.githubusercontent.com/202757/72247948-4e4b3500-35f6-11ea-9a58-b323bec1fb98.png width=45%><img src=https://user-images.githubusercontent.com/202757/72247949-4e4b3500-35f6-11ea-821d-02b32b8a265e.png width=45%>

**Comment by felipevaz** [comment]
Oh boy, I missed that one completely. Thanks!

**Comment by Frenzie** [comment]
Added in https://github.com/koreader/koreader/pull/3979 btw.

Some discussion on hyphenation and screenshots of bad justification here: https://github.com/koreader/crengine/pull/150#issuecomment-386858028 (It's probably exaggerated by disabling hyphenation, but part of the point is that most auto-hyphenation isn't that great, including in CREngine/KOReader.)
